packages/flutter/lib/src/cupertino/text_selection_toolbar.dart

```diff
@@ -395,12 +393,34 @@ class _RenderCupertinoTextSelectionToolbarShape extends RenderShiftedBox {
     if (child == null) {
       return;
     }
+
+    final BoxParentData childParentData = child.parentData! as BoxParentData;
+
+    final RRect rrect = _shapeRRect(child);
+    final Path clipPath = _clipPath(child, rrect);
+
+    // If configured, paint the shadow beneath the shape.
+    if (_shadowColor != null) {
+      final BoxShadow boxShadow = BoxShadow(
+        color: _shadowColor!,
+        blurRadius: 15.0,
+      );
+      final RRect shadowRRect = RRect.fromLTRBR(
+        rrect.left,
+        rrect.top,
+        rrect.right,
+        rrect.bottom + _kToolbarArrowSize.height,
+        _kToolbarBorderRadius,
+      ).shift(offset + childParentData.offset + boxShadow.offset);
+      context.canvas.drawRRect(shadowRRect, boxShadow.toPaint());
```

Comment:
have you tried `Canvas.drawShadow`? According to the documentation it's more efficient.

Reply:
I did and it looked horrible :). Didn't repaint in some cases after tapping buttons.